src/main/java/bio/terra/service/job/JobService.java

```diff
@@ -303,13 +306,20 @@ public List<JobModel> enumerateJobs(
       filter.addFilterFlightClass(FlightFilterOp.EQUAL, className);
     }
 
+    HashSet<String> jobIdSet = new HashSet<>(ListUtils.emptyIfNull(jobIds));
     List<FlightState> flightStateList;
     boolean canListAnyJob = checkUserCanListAnyJob(userReq);
     try {
       if (canListAnyJob) {
         flightStateList = stairway.getFlights(offset, limit, filter);
+        if (!jobIdSet.isEmpty()) {
```

Comment:
The code checks `.isEmpty()` here and below in `includeFlight()` but these checks are unnecessary. If the set is empty, then calling `contains()` will always be `false`, so there is no need to avoid using it.